common/src/main/java/com/microsoft/identity/common/AndroidCommonComponents.java

```diff
@@ -26,27 +26,55 @@
 
 import androidx.annotation.Nullable;
 
-import com.microsoft.identity.common.crypto.AndroidBrokerStorageEncryptionManager;
 import com.microsoft.identity.common.crypto.AndroidAuthSdkStorageEncryptionManager;
+import com.microsoft.identity.common.crypto.AndroidBrokerStorageEncryptionManager;
+import com.microsoft.identity.common.internal.cache.SharedPreferencesFileManager;
 import com.microsoft.identity.common.internal.net.cache.HttpCache;
+import com.microsoft.identity.common.internal.platform.AndroidDeviceMetadata;
+import com.microsoft.identity.common.internal.platform.DevicePopManager;
 import com.microsoft.identity.common.internal.util.ProcessUtil;
+import com.microsoft.identity.common.internal.util.SharedPreferenceLongStorage;
+import com.microsoft.identity.common.java.crypto.IDevicePopManager;
 import com.microsoft.identity.common.java.crypto.IKeyAccessor;
+import com.microsoft.identity.common.java.exception.ClientException;
 import com.microsoft.identity.common.java.interfaces.ICommonComponents;
+import com.microsoft.identity.common.java.platform.Device;
 import com.microsoft.identity.common.java.telemetry.ITelemetryCallback;
+import com.microsoft.identity.common.java.util.ClockSkewManager;
+import com.microsoft.identity.common.java.util.IClockSkewManager;
 import com.microsoft.identity.common.logging.Logger;
 
+import java.io.IOException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
 import lombok.NonNull;
 
+import static com.microsoft.identity.common.java.exception.ClientException.CERTIFICATE_LOAD_FAILURE;
+import static com.microsoft.identity.common.java.exception.ClientException.IO_ERROR;
+import static com.microsoft.identity.common.java.exception.ClientException.KEYSTORE_NOT_INITIALIZED;
+import static com.microsoft.identity.common.java.exception.ClientException.NO_SUCH_ALGORITHM;
+
 /**
  * Android implementations of platform-dependent components in Common.
  */
 public class AndroidCommonComponents implements ICommonComponents {
     private static String TAG = AndroidCommonComponents.class.getSimpleName();
 
+    /**
+     * SharedPref filename for Clock Skew storage.
+     */
+    private static final String SKEW_PREFERENCES_FILENAME =
+            "com.microsoft.identity.client.clock_correction";
+
     protected final Context mContext;
+    private IClockSkewManager mClockSkewManager;
+    private IDevicePopManager mDefaultDevicePoPManager;
 
-    public AndroidCommonComponents(@NonNull final Context context){
+    public AndroidCommonComponents(@NonNull final Context context) {
```

Comment:
I think the intention here is that this be Application context rather than activity context.  We may want to check that we're getting the kind of context that we expect or at least not getting a context that we don't expect.

Reply:
I see that below the clock skew manager and device pop manager are lazy loaded.  Do you want to instantiate those in the constructor instead?  I'm curious why lazy loading is being used in this case?  If something is wrong with initializing either of those when do you want to find out?